pkg/sql/logictest/testdata/logic_test/alter_table

```diff
@@ -3174,6 +3172,159 @@ table_name  constraint_name  constraint_type  details
+  CONSTRAINT t_99035_pkey PRIMARY KEY (j ASC),
+  UNIQUE INDEX t_99035_i_key (i ASC)
+)
+
+statement ok
+DROP TABLE IF EXISTS t_99035;
+CREATE TABLE t_99035 (i INT PRIMARY KEY, j INT NOT NULL, FAMILY "primary" (i, j));
+INSERT INTO t_99035 VALUES (0,0);
+
+skipif config local-legacy-schema-changer
+skipif config local-mixed-22.2-23.1
+statement ok
+ALTER TABLE t_99035 ALTER PRIMARY KEY USING COLUMNS (j), DROP COLUMN i;
+
+skipif config local-legacy-schema-changer
+skipif config local-mixed-22.2-23.1
+query T
+SELECT create_statement FROM [SHOW CREATE TABLE t_99035]
+----
+CREATE TABLE public.t_99035 (
+  j INT8 NOT NULL,
+  CONSTRAINT t_99035_pkey PRIMARY KEY (j ASC)
+)
+
+statement ok
+DROP TABLE IF EXISTS t_99035;
+CREATE TABLE t_99035 (i INT PRIMARY KEY, j INT NOT NULL, FAMILY "primary" (i, j));
+INSERT INTO t_99035 VALUES (0,0);
+
+skipif config local-legacy-schema-changer
+skipif config local-mixed-22.2-23.1
+statement ok
+ALTER TABLE t_99035 ADD COLUMN k INT DEFAULT unique_rowid(), ALTER PRIMARY KEY USING COLUMNS (j), DROP COLUMN i;
+
+skipif config local-legacy-schema-changer
+skipif config local-mixed-22.2-23.1
+query T
+SELECT create_statement FROM [SHOW CREATE TABLE t_99035]
+----
+CREATE TABLE public.t_99035 (
+  j INT8 NOT NULL,
+  k INT8 NULL DEFAULT unique_rowid(),
+  CONSTRAINT t_99035_pkey PRIMARY KEY (j ASC)
+)
+
+statement ok
+DROP TABLE IF EXISTS t_99035;
+CREATE TABLE t_99035 (i INT PRIMARY KEY, j INT NOT NULL, FAMILY "primary" (i, j));
+INSERT INTO t_99035 VALUES (0,0);
+
+skipif config local-legacy-schema-changer
+skipif config local-mixed-22.2-23.1
+statement error validation of CHECK "j > 0:::INT8" failed on row: j=0, k=[0-9]+, p=30
+ALTER TABLE t_99035 ADD COLUMN k INT DEFAULT unique_rowid(), ALTER PRIMARY KEY USING COLUMNS (j), DROP COLUMN i, ADD COLUMN p INT DEFAULT 30, ADD CHECK (j>0);
+
+query II
+SELECT * FROM t_99035
+----
+0 0
+
+skipif config local-legacy-schema-changer
+skipif config local-mixed-22.2-23.1
+statement ok
+ALTER TABLE t_99035 ADD COLUMN k INT DEFAULT unique_rowid(), ALTER PRIMARY KEY USING COLUMNS (j), DROP COLUMN i, ADD COLUMN p INT DEFAULT 30, ADD CHECK (j>=0);
+
+skipif config local-legacy-schema-changer
+skipif config local-mixed-22.2-23.1
+query T
+SELECT create_statement FROM [SHOW CREATE TABLE t_99035]
+----
+CREATE TABLE public.t_99035 (
+  j INT8 NOT NULL,
+  k INT8 NULL DEFAULT unique_rowid(),
+  p INT8 NULL DEFAULT 30:::INT8,
+  CONSTRAINT t_99035_pkey PRIMARY KEY (j ASC),
+  CONSTRAINT check_j CHECK (j >= 0:::INT8)
+)
+
+statement ok
+DROP TABLE IF EXISTS t_99035;
+CREATE TABLE t_99035 (i INT PRIMARY KEY, j INT NOT NULL, FAMILY "primary" (i, j));
+INSERT INTO t_99035 VALUES (0,0);
+
+skipif config local-legacy-schema-changer
+skipif config local-mixed-22.2-23.1
+statement ok
+ALTER TABLE t_99035 ADD COLUMN k INT DEFAULT unique_rowid(), ALTER PRIMARY KEY USING COLUMNS (j), DROP COLUMN i, ADD COLUMN p INT DEFAULT 0, ADD CHECK (j>=0), ADD FOREIGN KEY (p) REFERENCES t_99035(j);
```

Comment:
Add this as a schema changer end-to-end test case and make sure you exercise the DML injection test properly and I'll be happy.